diff --git a/node_modules/react-native-css-interop/dist/css-to-rn/parseDeclaration.js b/node_modules/react-native-css-interop/dist/css-to-rn/parseDeclaration.js
index 59ea66e..088149d 100644
--- a/node_modules/react-native-css-interop/dist/css-to-rn/parseDeclaration.js
+++ b/node_modules/react-native-css-interop/dist/css-to-rn/parseDeclaration.js
@@ -918,6 +918,7 @@ function parseDeclaration(declaration, options) {
             return addStyleProp(declaration.property, parseTextAlign(declaration.value, parseOptions));
         case "box-shadow": {
             parseBoxShadow(declaration.value, parseOptions);
+            return;
         }
         case "aspect-ratio": {
             return addStyleProp(declaration.property, parseAspectRatio(declaration.value));
diff --git a/node_modules/react-native-css-interop/src/css-to-rn/parseDeclaration.ts b/node_modules/react-native-css-interop/src/css-to-rn/parseDeclaration.ts
index bf095a5..c990bfd 100644
--- a/node_modules/react-native-css-interop/src/css-to-rn/parseDeclaration.ts
+++ b/node_modules/react-native-css-interop/src/css-to-rn/parseDeclaration.ts
@@ -1556,6 +1556,7 @@ export function parseDeclaration(
       );
     case "box-shadow": {
       parseBoxShadow(declaration.value, parseOptions);
+      return;
     }
     case "aspect-ratio": {
       return addStyleProp(
