diff --git a/node_modules/react-native-barcodes-detector/android/build.gradle b/node_modules/react-native-barcodes-detector/android/build.gradle
index e1123ba..9f9e053 100644
--- a/node_modules/react-native-barcodes-detector/android/build.gradle
+++ b/node_modules/react-native-barcodes-detector/android/build.gradle
@@ -5,7 +5,6 @@ buildscript {
   repositories {
     google()
     mavenCentral()
-    jcenter()
   }
 
   dependencies {
@@ -52,7 +51,6 @@ android {
 
 repositories {
   mavenCentral()
-  jcenter()
   google()
 
   def found = false
